File: typescript/src/models/paciente.ts
// telefono tambien se puede implementar como: type Telefono = { codigo:String; numero;String;}

import { Persona } from "./persona";
import { SubRegistro } from "./subRegistro";
import { Telefono } from "./telefono";

export class Paciente extends Persona {
    private edad: number;
    private profesion: String;

    private peso?: SubRegistro<number>;

    private correo: String;
    private telefonos: Telefono[];

    constructor(
        id: number,
        nombre: String,
        apellido: String,
        edad: number,
        profesion: String,
        correo: String,
        telefonos: Telefono[],
        peso?: SubRegistro<number>
    ) {
        super(id, nombre, apellido);
        this.edad = edad;
        this.profesion = profesion;
        this.peso = peso;
        this.correo = correo;
        this.telefonos = telefonos;
    }

    private setEdad(edad: number) {
        this.edad = edad;
    }

    public getEdad(): number {
        return this.edad;
    }

    private setProfesion(profesion: String) {
        this.profesion = profesion;
    }

    public getProfesion(): String {
        return this.profesion;
    }

    private setPeso(peso: SubRegistro<number>) {
        this.peso = peso;
    }

    // buscar la forma de que esto
    public getPeso(): SubRegistro<number> | undefined {
        return this.peso;
    }

    private setCorreo(correo: String) {
        this.correo = correo;
    }

    public getCorreo() {
        return this.correo;
    }

    public getTelefonos(): Telefono[] {
        return this.telefonos;
    }

    private setTelefonos(telefonos: Telefono[]) {
        this.telefonos = telefonos;
    }
}

File: dart/lib/src/models/paciente.dart
//import 'dart:svg';
import 'telefono.dart';
import 'persona.dart';
import 'subRegistro.dart';
import '/src/handlers/ManejadorBuscadorPaciente.dart';
import 'doctor.dart';

class Paciente extends Persona {
  int edad;
  String profesion;
  int peso;
  String correo;
  List<Telefono> telefonos;
  ManejadorBuscadorPaciente manejador;
  Paciente(int id, String nombre, String apellido, this.edad, this.profesion,
      this.peso, this.correo, this.telefonos, this.manejador)
      : super(id, nombre, apellido);

  int get getId {
    return super.id;
  }

  String get getNombre {
    return super.nombre;
  }

  String get getApellido {
    return super.apellido;
  }

  int get getEdad {
    return edad;
  }

  String get getProfesion {
    return profesion;
  }

  int get getPeso {
    return peso;
  }

  String get getCorreo {
    return correo;
  }

  List<Telefono> get getTelefonos {
    return telefonos;
  }

  void set setNombre(String nomb) {
    super.nombre = nomb;
  }

  void set setApellido(String ape) {
    super.apellido = ape;
  }

  void set setEdad(int ed) {
    this.edad = ed;
  }

  void set setEProfesion(String pro) {
    this.profesion = pro;
  }

  void set setCorreo(String cor) {
    this.correo = cor;
  }

  void setTelefonos(int numb, int posicion) {
    this.telefonos[posicion].numero = numb;
  }
}
